File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/auth/dto/LoginResponse.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.auth.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto.UserInfo;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users;

public record LoginResponse(UserInfo user,
                            String accessToken,
                            String refreshToken) {

    public static LoginResponse fromEntityAndTokens(Users user,
                                                    String accessToken,
                                                    String refreshToken) {

        return new LoginResponse(
                UserInfo.fromEntity(user),
                accessToken,
                refreshToken
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/auth/dto/LoginResponse.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.auth.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto.UserInfo
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.mapper.toInfo

data class LoginResponse(
    val user: UserInfo,
    val accessToken: String,
    val refreshToken: String
) {
    companion object {
        fun fromEntityAndTokens(user: Users, accessToken: String, refreshToken: String) =
            LoginResponse(
                user = user.toInfo(),
                accessToken = accessToken,
                refreshToken = refreshToken
            )
    }
}
